src/driver/oracle/OracleQueryRunner.ts

```diff
@@ -172,7 +172,7 @@ export class OracleQueryRunner extends BaseQueryRunner implements QueryRunner {
                         this.driver.connection.logger.logQueryError(err, query, parameters, this);
                         return fail(new QueryFailedError(query, parameters, err));
                     }
-                    ok(result.rows || result.outBinds);
+                    ok(result.rows || result.outBinds || result.rowsAffected);
```

Comment:
This approach doesn't look good. I suggest to add a third flag paramater to `query` method, something like `fullResult: boolean` (don't like this name however) and make a check here and return some `QueryResult` interface (need to create a new interface) with `rowsAffected` inside as well with all other additional data. Then you just call a method in `delete query runner` with fullResult = true.

Also, can you please implement same for other methods (update and insert)? Also would be really helpful if you can do it for all other drivers as well and we can officially support this feature and add documentation to it.